db/active_replicator_pull.go

```diff
@@ -40,10 +40,6 @@ func (apr *ActivePullReplicator) Start(ctx context.Context) error {
 	apr.lock.Lock()
 	defer apr.lock.Unlock()
 
-	if apr == nil {
-		return fmt.Errorf("nil ActivePullReplicator, can't start")
-	}
```

Comment:
Wouldn't a more appropriate fix be to move this to above the lock, or is it guaranteed that `apr` can't be nil?

Reply:
I think because the code works as is, that this was just overly defensive programming and broken in a refactor.